documentation/modules/adding-source-provider.adoc

```diff
@@ -26,37 +26,166 @@ You can add {a-rhv} source provider by using the {ocp} web console.
 . Specify the following fields:
 
-* *Provider name*: Name to display in the list of providers
-* *vCenter host name or IP address*: vCenter host name or IP address - if a certificate for FQDN is specified, the value of this field needs to match the FQDN in the certificate
-* *vCenter user name*: vCenter user, for example, `user@vsphere.local`
-* *vCenter password*: vCenter user password
-* *VDDK init image*: VDDKInitImage path
+* *Provider resource name*: Name to display in the list of providers.
+* *URL*: Endpoint.
+* *VDDK init image*: `VDDKInitImage` path. It is strongly recommended to create a VDDK init image to accelerate migrations. For more information, see xref:../master.adoc#creating-vddk-image_mtv[Creating a VDDK image].
+* *Username*: vCenter user, for example, `user@vsphere.local`.
+* *Password*: vCenter user password.
+* *SHA-1 fingerprint*: The provider currently requires the SHA-1 fingerprint of the vCenter Server's TLS certificate in all circumstances. vSphere calls this the server's thumbprint.
++
+
+. Choose one of the following options for validating CA certificates:
+
+** *Skip certificate validation* : Migrate without validating a CA certificate.
+** *Use the system CA certificates*: Migrate after validating the system CA certificates.
 
-. To allow a migration without validating the provider's CA certificate, select the *Skip certificate validation* check box. By default, the checkbox is cleared, meaning that the certificate will be validated.
-. Enter the *SHA-1 fingerprint*.
+.. To skip certificate validation, select the *Skip certificate validation* check box.
+.. To validate the system CA certificates, leave the *Skip certificate validation* check box cleared.
 endif::[]
 ifdef::rhv[]
-. Select *Red Hat Virtualization* from the *Provider type* list.
+. Click *Red Hat Virtualization*
 . Specify the following fields:
 
-* *Provider name*: Name to display in the list of providers
-* *RHV Manager host name or IP address*: {manager} host name or IP address -  if a certificate for FQDN is specified, the value of this field needs to match the FQDN in the certificate
-* *RHV Manager user name*: {manager} user
-* *RHV Manager password*: {manager} password
+* *Provider resource name*: Name to display in the list of providers
+* *URL*: API endpoint of the {rhv-short} Manager on which the source VM is mounted
+* *Username*: Username
+* *Password*: Password
+
+. Choose one of the following options for validating CA certificates:
+
+** *Skip certificate validation* : Migrate without validating a CA certificate.
+** *Use the system CA certificates*: Migrate after validating the system CA certificates.
+** *Use a custom CA certificate*: Migrate after validating a custom CA certificate.
+
+.. To skip certificate validation, select the *Skip certificate validation* check box.
+.. To validate the system CA certificates, leave the *Skip certificate validation* check box cleared and leave the *CA certificate* text box empty.
```

Comment:
@qiyuann did you add the certificate to the system pool (to the node)? AFAIR @eslutsky verified this functionality with OpenStack, and the mechanism for RHV is similar

Reply:
@qiyuann hi, i'm currently reviewing this text and saw this question,
after flowing @ahadas instructions, does it work ?

maybe we need to document that we need to:
> did you add the certificate to the system pool (to the node)?

Reply:
@ahadas Do you mean to add the RHV certificate to /etc/pki/ca-trust/source/anchors/ on the node, and run update-ca-trust? I did so, but still can't create RHV provider with empty cacert.

Reply:
@qiyuann ok, the certificate authorities don't propagate from the node as I thought, and it's not that simple to add certificates to the system pol, see: https://docs.openshift.com/container-platform/4.13/networking/configuring-a-custom-pki.html
so yeah, even though we have the ability to consume system certificates, if this only applies to certificates that are built in into the image then there's not much value in it and we probably shouldn't document it...

Reply:
> and we probably shouldn't document it...

I'm not sure about that one actually -
on the one hand, the code supports it and it seems that with a cluster proxy we can provide a custom CA but on the other hand, we didn't test it properly and we'll get reports on this functionality..
when we have three options: not to validate the certificate, customer certificate and system certificate - it's easy to drop the last one
when there are only two, not to validate the certificate and system certificate, for vsphere - it's more problematic but if in practice we didn't test the connectivity with a certificate, and I understand it has not been verified before MTV as well, then it looks like we better drop it this option and instruct to always skip the certificate check..
@qiyuann @yaacov what do you think?

Reply:
@ahadas From a documentation standpoint, I can say the feature is for future use and users should select the checkbox BUT I'm not sure how great an idea that is, especially since we have this feature or something close to it in 2.4. https://access.redhat.com/documentation/en-us/migration_toolkit_for_virtualization/2.4/html-single/installing_and_using_the_migration_toolkit_for_virtualization/index#migrating-vms-web-console

Maybe instead, leave the documentation and add some type of note saying it might not work in some cases (help me with the details, if this sounds correct). @anarnold97  keep me honest, please...

Reply:
@RichardHoch you're correct that this also applies to vSphere providers in MTV 2.4
so maybe we can just drop this option for RHV..